### PR TITLE
Add new `vue/enforce-style-attribute` rule

### PR DESCRIPTION
**Pull request by mussinbenarbia**
This PR adds a new rule as proposed in [this issue](https://github.com/vuejs/eslint-plugin-vue/issues/2109).

### CODE REVIEW

**Review by ota-meshi — CHANGES_REQUESTED** [review]
See https://github.com/vuejs/eslint-plugin-vue/issues/2109#issuecomment-1479055924

**Comment by mussinbenarbia** [comment]
@ota-meshi Sorry for the very long absence 🙏 I made some modifications to the code to mimick how it's done in the `eslint-plugin-vue-scoped-css` plugin. Could you please take a look at the implementation? I have not updated the docs as I would like to make sure you agree with the implementation first.

Thank you 🙇

File: docs/rules/enforce-style-attribute.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+> enfore either the `scoped` or `module` attribute in SFC top level style tags
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+
+## :wrench: Options
+
+```json
+{
+  "vue/attribute-hyphenation": ["error", "either" | "scoped" |  "module"]
```
Comment (mussinbenarbia) [inline docs/rules/enforce-style-attribute.md:18]: issue: wrong rule name, need to update once implementation is finalized

**Comment by ota-meshi** [comment]
That rule sounds good to me. Could you undraft this PR when it is ready for review?

**Review by ota-meshi — CHANGES_REQUESTED** [review]
That rule sounds good to me. Could you undraft this PR when it is ready for review?

**Comment by mussinbenarbia** [comment]
@ota-meshi Thank you for taking a look! I will update the docs to reflect the new options and mark it as ready for review.

**Comment by mussinbenarbia** [comment]
@ota-meshi (Looping in @FloEdelmann as well) I updated the docs but I have a question. When I run `docs:watch` or `docs:build`, the following piece of text gets automatically added to my doc file for the rule:

```
- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
```

I'm not sure why. This rule does not have a fixer at the moment, so that paragraph is not required but it keeps being added automatically. Is it ok to remove it and push the code? Is that not going to make the CI fail?

**Comment by mussinbenarbia** [comment]
Also, I'm now wondering if the `allows` option should actually be singular form `allow` to keep some form of consistency with other rules such as [no-shadow](https://github.com/eslint/eslint/blob/main/docs/src/rules/no-shadow.md) 🤔 Do you agree?

File: lib/rules/enforce-style-attribute.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        'enforce or forbid the use of the `scoped` and `module` attributes in SFC top level style tags',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/enforce-style-attribute.html'
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          allows: {
```
Comment (FloEdelmann) [inline lib/rules/enforce-style-attribute.js:60]: I agree that it should be called `allow` instead, for consistency with other rules.

File: lib/rules/enforce-style-attribute.js [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+module.exports = {
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description:
+        'enforce or forbid the use of the `scoped` and `module` attributes in SFC top level style tags',
+      categories: undefined,
+      url: 'https://eslint.vuejs.org/rules/enforce-style-attribute.html'
+    },
+    fixable: 'code',
```
[the remark this answers is not in the document]
Comment (FloEdelmann) [reply lib/rules/enforce-style-attribute.js]: Sorry for the double work; I just checked it out myself already. See

* https://github.com/vuejs/eslint-plugin-vue/pull/2343

Thanks for checking though :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (mussinbenarbia) [reply lib/rules/enforce-style-attribute.js]: No problem! Then I'll go ahead and resolve this 👍

File: docs/rules/enforce-style-attribute.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+</eslint-code-block>
+
+## :wrench: Options
+
+```json
+{
+  "vue/enforce-style-attribute": [
+    "error",
+    { "allow": ["scoped", "module", "no-attributes"] }
```
Comment (ota-meshi) [inline docs/rules/enforce-style-attribute.md:75]: Hmm. I think the name `no-attributes` is misleading. lang="scss" has attributes. What do you think?

**Review by ota-meshi — COMMENTED** [review]
Sorry for the late review.

File: docs/rules/enforce-style-attribute.md (hunk 4 above)
Comment (mussinbenarbia) [reply docs/rules/enforce-style-attribute.md:75]: Yeah... that could be confusing 🤔 Maybe using `plain` like in the other plugin is the best choice here. If you agree I'll rename it

Comment (ota-meshi) [reply docs/rules/enforce-style-attribute.md:75]: I think `plain` is better :+1:

**Comment by mussinbenarbia** [comment]
@ota-meshi I renamed the option to `plain`. I pushed the code but it seems some tests for other unrelated rules are failing. I would appreciate if you could take another look 🙇

**Comment by ota-meshi** [comment]
Yeah. CI error should be fixed by #2357. So ignore it for now.

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**